src/test/java/space/obminyashka/items_exchange/controller/AuthorizationControllerTest.java

```diff
@@ -71,4 +88,127 @@ void register_whenMailServiceFailed_shouldThrowEmailSendingException() {
                 () -> verify(mailService).sendEmailTemplateAndGenerateConfrimationCode(eq(dto.getEmail()), eq(EmailType.REGISTRATION), any()),
                 () -> verifyNoMoreInteractions(userService));
     }
+
+    @Test
+    void login_shouldLoginUser() {
+        var expectedUserDto = new UserLoginDto();
+        expectedUserDto.setUsernameOrEmail("user");
+        var prefinalizeUserLoginDto = createUserLoginDto();
+        var expectedUserLoginDto = finalizeUserLoginDto(prefinalizeUserLoginDto);
+        when(userService.findAuthDataByUsernameOrEmail(any())).thenReturn(prefinalizeUserLoginDto);
+        when(authService.finalizeAuthData(any())).thenReturn(expectedUserLoginDto);
+
+        var actualUserLoginDto = authController.login(expectedUserDto);
+
+        assertAll(
+                () -> verify(userService).findAuthDataByUsernameOrEmail(usernameCaptor.capture()),
+                () -> assertEquals(expectedUserLoginDto.getUsername(), usernameCaptor.getValue()),
+                () -> verify(authService).finalizeAuthData(authCaptor.capture()),
+                () -> assertEquals(prefinalizeUserLoginDto, authCaptor.getValue()),
+                () -> verify(authenticationManager).authenticate(any()),
+                () -> assertThat(actualUserLoginDto)
+                        .hasFieldOrPropertyWithValue("username", expectedUserLoginDto.getUsername())
+                        .hasFieldOrPropertyWithValue("email", expectedUserLoginDto.getEmail())
+                        .hasFieldOrPropertyWithValue("firstName", expectedUserLoginDto.getFirstName())
+                        .hasFieldOrPropertyWithValue("lastName", expectedUserLoginDto.getLastName())
+                        .hasFieldOrPropertyWithValue("language", expectedUserLoginDto.getLanguage())
+                        .hasFieldOrPropertyWithValue("role", expectedUserLoginDto.getRole())
+                        .hasFieldOrPropertyWithValue("accessToken", expectedUserLoginDto.getAccessToken())
+                        .hasFieldOrPropertyWithValue("accessTokenExpirationDate", expectedUserLoginDto.getAccessTokenExpirationDate())
+                        .hasFieldOrPropertyWithValue("refreshToken", expectedUserLoginDto.getRefreshToken())
+                        .hasFieldOrPropertyWithValue("refreshTokenExpirationDate", expectedUserLoginDto.getRefreshTokenExpirationDate())
+                        .hasFieldOrPropertyWithValue("avatarImage", expectedUserLoginDto.getAvatarImage())
```

Comment:
Why don't you simply compare two objects using equals/hashcode instead of making cooperation field by field?